pallets/migrations/src/lib.rs

```
@@ -208,18 +224,20 @@ pub mod pallet {
 					"invoking post_upgrade() on migration {}", migration_name
 				);
 
-				let result = migration.post_upgrade();
-				match result {
-					Ok(()) => {
-						log::info!("migration {} post_upgrade() => Ok()", migration_name);
-					}
-					Err(msg) => {
-						log::error!(
-							"migration {} post_upgrade() => Err({})",
-							migration_name,
-							msg
-						);
-						failed = true;
+				if let Some(state) = migration_states_map.get(&migration_name.to_string()) {
```

It might be better to fail hard with an `expect()` here or similar